java/algorithm/src/main/java/org.apache.arrow.algorithm.sort/DefaultVectorComparators.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.arrow.algorithm.sort;
+
+import org.apache.arrow.memory.BufferAllocator;
+import org.apache.arrow.vector.IntVector;
+import org.apache.arrow.vector.VarCharVector;
+import org.apache.arrow.vector.holders.NullableVarCharHolder;
+
+/**
+ * Default comparator implementations for different types of vectors.
+ */
+public class DefaultVectorComparators {
+
+  /**
+   * Default comparator for 32-bit integers.
+   * The comparison is based on int values, with null comes first.
+   */
+  public static class IntComparator extends VectorValueComparator<IntVector> {
+
+    public IntComparator() {
+      super(4);
+    }
+
+    @Override
+    public IntVector newVector(BufferAllocator allocator) {
+      return  new IntVector("", allocator);
```

Comment:
wierd spacing?

Reply:
Thanks for the careful review. This code is removed in the new commit.